### PR TITLE
Set scoped omp to fix IVF index build degrade

### PR DESCRIPTION
**Pull request by cydrain**
issue: #861

### CODE REVIEW

File: benchmark/hdf5/benchmark_knowhere.h [hunk 1]
```diff
@@ -27,6 +27,9 @@
 #include "knowhere/index/index_factory.h"
 #include "knowhere/version.h"
 
+static const size_t default_build_thread_num = 8;
```
Comment (alexanderguzhva) [inline benchmark/hdf5/benchmark_knowhere.h:30]: https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency ? Please don't forget to check for 0 if you decide to go this path.

Comment (cydrain) [reply benchmark/hdf5/benchmark_knowhere.h:30]: Hi Alex, this variable is only used in benchmark

**Review by Presburger — APPROVED** [review]

File: src/common/thread/thread.cc [hunk 2]
```diff
@@ -72,9 +72,14 @@ GetBuildThreadPoolSize() {
     return ThreadPool::GetGlobalBuildThreadPool()->size();
 }
 
-std::unique_ptr<ThreadPool::ScopedOmpSetter>
-CreateScopeOmpSetter(int num_threads) {
-    return std::make_unique<ThreadPool::ScopedOmpSetter>(num_threads);
+std::unique_ptr<ThreadPool::ScopedBuildOmpSetter>
```
[the remark this answers is not in the document]
Comment (cydrain) [reply src/common/thread/thread.cc:75]: this API will be removed later

**Pull request merged**